externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala

```diff
@@ -64,7 +82,7 @@ object SchemaHelper {
     case _: ArrayType => TTypeId.ARRAY_TYPE
     case _: MapType => TTypeId.MAP_TYPE
     case _: StructType => TTypeId.STRUCT_TYPE
-    // TODO: it is private now, case udt: UserDefinedType => TTypeId.USER_DEFINED_TYPE
+    case dt: DateType if isUserDefinedType(dt) => TTypeId.USER_DEFINED_TYPE
```

Comment:
Is case udt: UserDefinedType => TTypeId.USER_DEFINED_TYPE good to go？

BTW, you shall not use `DateType` but `DataType` here.

Reply:
Thanks, i will revise the dataType. Besides, UserDefinedType is still private in spark 3.1, so we may avoid importing this class.